File: app/src/main/java/ardjomand/leonardo/nutrimeal/cart/EditCartMealContract.java
package ardjomand.leonardo.nutrimeal.cart;

import ardjomand.leonardo.nutrimeal.data.pojos.CartMeal;

interface EditCartMealContract {

    interface View {

        void updateCartMeal(CartMeal meal);
    }

    interface Presenter {

        void setView(View view);

        void subscribe(String key);

        void unsubscribe();

        void increaseQuantity();

        void decreaseQuantity();
    }
}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/cart/EditCartMealContract.kt
package ardjomand.leonardo.nutrimeal.cart

import ardjomand.leonardo.nutrimeal.data.pojos.CartMeal

internal interface EditCartMealContract {

    interface View {

        fun updateCartMeal(cartMeal: CartMeal)
    }

    interface Presenter {

        fun setView(view: View?)

        fun subscribe(key: String)

        fun unsubscribe()

        fun increaseQuantity()

        fun decreaseQuantity()
    }
}
